docs/addons/calico.md

```diff
@@ -0,0 +1,197 @@
+```
+### Applying Policies using Calico to block traffic
+
+One of the powerful things about Calico is the ability to apply a Global Network Policy - policy across all namespaces.
+
+Let's apply the following Global Network Policy:
+
+```yaml
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: global-deny-all
+spec:
+  order: 2000
+  types:
+  - Ingress
+  - Egress
+
+  # egress network rules
+  egress:
+  # Allow all egress traffic from kube-system.
+  - action: Allow
+    destination: {}
+    source:
+      namespaceSelector: name == 'kube-system'
+
+  # Allow egress DNS traffic to any destination.
+  - action: Allow
+    protocol: UDP
+    destination:
+      nets:
+        - 0.0.0.0/0
+      ports:
+        - 53
+
+  # ingress network rules
+  ingress:
+  # Allow all ingress traffic for the kube-system namespace.
+  - action: Allow
+    destination:
+      namespaceSelector: name == 'kube-system'
+    source: {}
+```
+
+You can either save it as `calico-global-deny-all.yaml` or copy the file in this directory. Run the following command to apply the policy:
+
+```bash
+DATASTORE_TYPE=kubernetes \
+    KUBECONFIG=~/.kube/config
+    calicoctl apply -f calico-global-deny-all.yaml 
+```
+
+You should see the following:
+
+```
+Successfully applied 1 'GlobalNetworkPolicy' resource(s)
+```
+
+The policy blocks all egress and ingress traffic in namespaces except `kube-system`, with the exception of DNS traffic (UDP:53). 
+You can try to ping the pod from `team-burnham` from a shell within the pod from `team-riker` and see that now you won't receive any response back.
+
+### Applying additional policy to re-open pod to pod communications
+
+The above GlobalNetworkPolicy removes the ability for pods to communicate with other pods in all namespaces except `kube-system`. 
+However, you can still apply Kubernetes NetworkPolicy on top of that to “poke holes” for egress and ingress needs. 
+
+For example, if you wanted to open all traffic between `team-burnham` and `team-riker`, the following Kubernetes NetworkPolicy could be applied. This policy allows any endpoint (i.e. pods) from `team-riker` namespace to make ping requests (ICMP) to any endpoint in `team-burnham` namespace.
+
+```yaml
+# This GlobalNetworkPolicy uses Calico's CRD
+# (https://docs.projectcalico.org/v3.5/reference/calicoctl/resources/globalnetworkpolicy)
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: allow-burnham-riker
+spec:
+  namespaceSelector: name == 'team-burnham'
+  order: 3000
+  types:
+    - Ingress
+
+  # ingress network rules
+  ingress:
+    # Allow all ingress traffic from team-riker to team-burnham namespaces
+    - action: Allow
+      protocol: ICMP
+      source:
+        namespaceSelector: name == 'team-riker'
+      icmp:
+        type: 8
+
+```
+
+Apply the new GlobalNetworkPolicy:
+
+```bash
+calicoctl apply -f calico-allow-burnham-riker.yaml     
```

Comment:
Since this resource is using CRD, do we really need calicoctl to apply this to the cluster? What happens if we just use `kubectl apply -f` in this case?

Reply:
replaced this with k8s network policy. It uses kubectl now.